salt/modules/file.py

```diff
@@ -4315,7 +4321,21 @@ def source_list(source, source_hash, saltenv):
                         ret = (single_src, single_hash)
                         break
                 elif proto.startswith("http") or proto == "ftp":
-                    query_res = salt.utils.http.query(single_src, method="HEAD")
+                    # If rquests is available we'll use that for the HEAD request in case the URL ends up being a
```

Comment:
What about for users that do not have requests installed by default? Will they still continue to get the traceback in the issue this PR resolves? Is there a way to catch this 3xx redirect is occurring and return an error letting the user know they should use requests?

Reply:
Yeah, the `decode_body=False` part of the `else` clause fixes it when not using requests. I just patched my local salt and it worked with just

```diff
-query_res = salt.utils.http.query(single_src, method="HEAD")
+query_res = salt.utils.http.query(single_src, method="HEAD", decode_body=False)
```

But I guess the rest of the PR doesn't hurt? Other than the part where throwing more details in delays its own merging.

Reply:
thanks for testing that @dequis . So it seems we do not necessarily need requests, but just need to pass in `decode_body=False`. Does that align with your findings @garethgreenaway ?

Reply:
bump @garethgreenaway ^

Reply:
Apologies for the delay on this one.  I was able to verify that including `decode_body=False)` in the call to salt.utils.http.query fixes it with both `requests` and `torando`.  I've updated the PR to include the changes.